syntax = "proto3";

package database;

import "types.proto";


message User {
    uint64 id = 1;

    string username = 2;
    string password = 3;

    string name = 4;
    uint64 score = 5;

    uint64 session_id = 6;

    uint64 queue_id = 7;
    uint64 game_id = 8;
}


// Suggested

message Queue {
    uint64 id = 1;
    uint32 size = 2;
    repeated uint64 user_ids = 3;
}


message Player {
    uint64 user_id = 1;
    types.Player block_info = 2;
}

message Game {
    uint64 id = 1;

    repeated Player players = 2;
    repeated types.FoodBlock foods = 3;
    repeated types.WallBlock walls = 4;
}

